rudder-core/src/main/scala/com/normation/rudder/services/nodes/NodeInfoService.scala

```diff
@@ -111,84 +111,97 @@ object NodeInfoServiceImpl {
+    /**
+     * Check if node related infos are up to date.
+     *
+     * Here, we need to only check for attributeModifyTimestamp in
+     * node from ou=Nodes (in Rudder and and ou=AcceptedInventories)
+     * Reason:
+     * - for inventories, any new inventory change the "receiveDate" attribute,
+     *   so the node entry is modified each time, even if there is no modification
+     *   or only modification in related entries (machine, etc).
+     *   The only other inventory changes are acceptation/deletion (they modify entry).
+     *
+     * - for ou=Node, all modifications happen in the entry, so the modifyTimestamp
+     *   is changed accordingly.
+     *
+     * Moreover, it is less costly to only do one search and post-filter result
+     * than to do 2.
+     *
+     * A cleaner implementation could use a two persistent search which would notify
+     * when a cache becomes invalide and reset it, but the rationnal for that implementation is:
+     * - it's extremelly simple to understand the logic (if(cache is uptodate) use it else update cache)
+     * - most of the time (99.99% of it), the search will return 0 result and will be cache on OpenLDAP,
+     *   whatever the number of entries. So we talking of a request taking a couple of ms on the server
+     *   (with a vagrant VM on the same host (so, almost no network), it takes from client to server and
+     *   back ~10ms on a dev machine.
      */
-    import scala.collection.mutable.{Map => MutMap}
+    def isUpToDate(lastKnowModification: DateTime): Boolean = {
+      val n0 = System.currentTimeMillis
+      val res = ldap.map { con =>
+        con.search(
+            nodeDit.BASE_DN
+          , Sub
+          , AND(
+                OR(
+                    // ou=Nodes,cn=rudder-configuration - the objectClass is used only here
+                    IS(OC_RUDDER_NODE)
+                    // ou=Nodes,ou=Accepted Inventories,ou=Inventories,cn=rudder-configuration
+                    // add a constraint on entryDN (onelevel below base accepted node) - LDAP
+                    // expert would become mad one that one
+                  , AND(IS(OC_NODE), Filter.create(s"entryDN:dnOneLevelMatch:=${inventoryDit.NODES.dn.toString}"))
+                )
+              , GTEQ(A_MOD_TIMESTAMP, GeneralizedTime(lastKnowModification.plusMillis(1)).toString)
+            )
+          , "1.1"
+        )
+      } match {
+        case Full(seq) =>
+          //filter DN only in ou=Nodes,ou=Accepted Inventories,ou=Inventories,cn=rudder-configuration
+          // or in ou=Nodes,cn=rudder-configuration
+          val res = !seq.exists { e => e.dn.getParent == nodeDit.NODES.dn || e.dn.getParent == inventoryDit.NODES.dn }
+          logger.trace(s"Cache check for node info gave '${res}' (${seq.size} entry returned)")
+          res
+        case _        => false
```

Comment:
Not a warning, because if it's because the server is down, there will be already other messages (from connection pool and other) and it will be just noise. But a debug message to understand why the cache is not use, why not.